### PR TITLE
fix: correctly handle multiple root certificates on Azure Stack Hub windows nodes

### PR DESCRIPTION
**Pull request by haofan-ms**
**Reason for Change**:

Update external cloud provider logic for windows node to retrieve root certificate from SSL certificate of ARM endpoint

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Review by jadarsie — APPROVED** [review]

File: parts/k8s/kuberneteswindowssetup.ps1 [hunk 1]
```diff
@@ -450,20 +450,33 @@ try
...
+            try { $webRequest.GetResponse() } catch {}
+            if (($null -eq $webRequest.ServicePoint) -Or ($null -eq $webRequest.ServicePoint.Certificate)) {
+                throw "SSL Certificate of ARM endpoint is null"
             }
-            $azsARMUri = [System.Uri]$azsJson.managementPortalURL
-            $azsRootCert = Get-ChildItem -Path Cert:\LocalMachine\Root | Where-Object {$_.DnsNameList.Unicode -contains $azsARMUri.Host.Substring($azsARMUri.Host.IndexOf(".")).TrimStart(".")}
+            $sslCert = $webRequest.ServicePoint.Certificate
+            $sslCertChain = New-Object -TypeName System.Security.Cryptography.X509Certificates.X509Chain
+            $sslCertChain.build($sslCert)
+            $sslRootCert = @($sslCertChain.ChainElements.Certificate)[-1]
```
Comment (jsturtevant) [inline parts/k8s/kuberneteswindowssetup.ps1:471]: Will the this always be the right ssl cert?

File: parts/k8s/kuberneteswindowssetup.ps1 [hunk 2]
```diff
@@ -450,20 +450,33 @@ try
...
             }
             $azsJson = Get-Content -Raw -Path $azsConfigFile | ConvertFrom-Json
-            if ([string]::IsNullOrEmpty($azsJson.managementPortalURL)) {
-                throw "managementPortalURL is empty, cannot get Azure Stack ARM uri"
+            if ([string]::IsNullOrEmpty($azsJson.resourceManagerEndpoint)) {
+                throw "resourceManagerEndpoint is empty, cannot get Azure Stack ARM uri"
+            }
+            $azsARMUri = [System.Uri]$azsJson.resourceManagerEndpoint
+            $webRequest = [Net.WebRequest]::Create($azsARMUri.AbsoluteUri)
+            try { $webRequest.GetResponse() } catch {}
```
Comment (jsturtevant) [inline parts/k8s/kuberneteswindowssetup.ps1:464]: I guess the response doesn't matter here.  All that matters is that the Certificate is retrieved which is checked on the next line so if there was no connectivity that would be null.

**Review by jsturtevant — COMMENTED** [review]
I don't have a ton of context here but lgtm.  Left a few non-blocking comments

File: parts/k8s/kuberneteswindowssetup.ps1 (hunk 1 above)
Comment (haofan-ms) [reply parts/k8s/kuberneteswindowssetup.ps1:471]: Yes, the root cert will be at the bottom of the cert chain

File: parts/k8s/kuberneteswindowssetup.ps1 (hunk 2 above)
Comment (haofan-ms) [reply parts/k8s/kuberneteswindowssetup.ps1:464]: That's correct, the purpose of web request is to get the ssl cert & cert chain of the ARM endpoint

**Review by jsturtevant — APPROVED** [review]

**Pull request merged**